lib/demos-signal-completion.c

```diff
@@ -0,0 +1,33 @@
+/**
+ * Executable version of demos-sch, which allows for simple usage from shell scripts.
+ *
+ * If using `init: yes` in config file, call this after script init, and then call again
+ * whenever you want to signal completion for current window.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+
+int main()
+{
+    char *str = getenv("DEMOS_PARAMETERS");
+    if (!str) {
+        return 100;
+    }
+
+    int fd_completed, fd_new_period;
+    if (sscanf(str, "%d,%d,", &fd_completed, &fd_new_period) != 2) {
+        return 101;
+    }
+
+    uint64_t buf = 1;
+    if (write(fd_completed, &buf, sizeof(buf)) != sizeof(buf)) {
+        return 1;
+    }
+    if (read(fd_new_period, &buf, sizeof(buf)) != sizeof(buf)) {
+        return 2;
+    }
```

Comment:
No - this binary is intended both for signalling initialization and completion, and the process library has 2 different calls for it (`demos_completed` and `demos_initialization_completed`). The library would have to be modified to support this, which I don't think is a good idea.

Reply:
I don't understand why to have one program for two things, without the ability to control which think it actually is. If this is just a quick hack, then OK, but if you want to use it for serious testing, then you may think you test completion, but instead you'll test initialization and vice versa. Why not to have two programs - one calling `demos_init(); demos_initialization_completed()` and second `demos_init(); demos_completed()`?

Additionally, it's better to use the library code so that it will receive more testing.

Reply:
1) It mostly is a quick hack - originally, I intended it for some of my own experiments and the automated tests, so that we don't need to create a new binary for each test case.
2) I understand what you mean, but currently, it is not possible to have 2 separate programs, because from the scheduler side, both signals are the same - `demos_initialization_completed()` does some checks and then internally calls `demos_completed()`. We could send a different signal for each, but that would require changes in multiple parts of the codebase and isn't needed except for this one script.
3) I agree it's not great, but I don't expect this to get enough usage internally or externally to justify the changes required to implement 2 separate binaries. What do you think?